src/components/Editor/Preview/index.js

```diff
@@ -47,10 +47,24 @@ class Preview extends PureComponent {
     self.onMouseDown = this.onMouseDown.bind(this);
   }
 
+  // To stop newly opened sources from inheriting the scroll-state
+  // of the currently open tab we force a scroll to top.
+  topScrollNewTab(dispSource) {
+    if (dispSource.get("loadedState") == "loaded") return;
+
+    const sourceLocation = dispSource.location;
+    if (!sourceLocation || !sourceLocation.line) {
+      sourceLocation = { line: 0, column: 0 };
+    }
+  }
+
   componentDidMount() {
     const { codeMirror } = this.props.editor;
+    const { selectedSource } = this.props;
     const codeMirrorWrapper = codeMirror.getWrapperElement();
 
+    this.topScrollNewTab(selectedSource);
```

Comment:
why is it in Preview/index and not Editor/index?

I would have expected this kind of logic in an action

Reply:
It is on the preview because the componentDidMount action occurs once for every tab mounted. In editor it would have to be in componentDidUpdate which seemed the less efficient place to put it in due to all the checks involved.

Sure, will put it in an action.